src/main/java/vg/civcraft/mc/civmodcore/world/ChunkTracker.java

```diff
@@ -0,0 +1,53 @@
+package vg.civcraft.mc.civmodcore.world;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.bukkit.Chunk;
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.world.ChunkLoadEvent;
+import org.bukkit.event.world.ChunkUnloadEvent;
+import vg.civcraft.mc.civmodcore.api.LocationAPI;
+
+/**
+ * This tracker should provide an efficient way of only using chunks that are loaded in.
+ */
+public class ChunkTracker implements Listener {
+
+	private static final Map<WorldXZ, Chunk> STORAGE = new HashMap<>();
```

Comment:
Replicating the state tracking that Bukkit already has seems a bit excessive for chunks. For worlds it's such a small amount that I don't really care, though it also kinda replicates logic and introduces additional area for bugs.

Reading from `updatingChunks` in PlayerChunkMap in ChunkProviderServer in WorldServer should do what you need.

Reply:
Removed ChunkTracker